src/lsst_efd_client/efd_helper.py

```diff
@@ -15,129 +18,188 @@
-        `aioinflux.client.InfluxDBClient`. The intent is to be able to
-        substitute a mocked client for testing.
-    """
 
-    influx_client = None
-    """The `aioinflux.client.InfluxDBClient` used for queries.
+class InfluxConnection(ABC):
+    """Abstract class to handle connections and basic querying
 
-    This should be used to execute queries not wrapped by this class.
+    Parameters
+    ----------
+    client : `aioinflux.client.InfluxDBClient`
+        An instance of a class `aioinflux.client.InfluxDBClient`.
+    mode: `ClientMode`
+        Client mode, async or sync.
+    auth: `lsst_efd_client.auth_helper.NotebookAuth`
+        Influxdb connection authorization details class instance.
+    db_name: `str`
+        Database name.
+    schema_registry_url: `str`
+        URL of the schema.
     """
 
-    subclasses = {}
-    deployment = ""
-
     def __init__(
         self,
-        efd_name,
-        db_name="efd",
-        creds_service="https://roundtable.lsst.codes/segwarides/",
-        timeout=900,
-        client=None,
+        client: aioinflux.InfluxDBClient,
+        auth: NotebookAuth,
+        db_name: str,
+        schema_registry_url: str,
     ):
-        self.db_name = db_name
-        self.auth = NotebookAuth(service_endpoint=creds_service)
-        (
-            host,
-            schema_registry_url,
-            port,
-            user,
-            password,
-            path,
-        ) = self.auth.get_auth(efd_name)
-        self.schema_registry_url = schema_registry_url
-        if client is None:
-            health_url = urljoin(f"https://{host}:{port}", f"{path}health")
-            response = requests.get(health_url)
-            if response.status_code != 200:
-                raise RuntimeError(
-                    f"InfluxDB server is not ready. "
-                    f"Recieved code:{response.status_code} "
-                    f"when reaching {health_url}."
-                )
-            self.influx_client = aioinflux.InfluxDBClient(
-                host=host,
-                path=path,
-                port=port,
-                ssl=True,
-                username=user,
-                password=password,
-                db=db_name,
-                mode="async",
-                output="dataframe",
-                timeout=timeout,
+        super().__init__()
+        self._client = client
+        self._auth = auth
+        self._db_name = db_name
+        self._schema_registry_url = schema_registry_url
+        self._query_history = []
+
+    @staticmethod
+    def create_client(client, mode, auth, db_name, schema_registry_url):
+        """
+        Creates a wrap for the influxdb client, sync or async,
+        depending on mode
+
+        Parameters
+        ----------
+        client : `aioinflux.client.InfluxDBClient`
+            An instance of a class `aioinflux.client.InfluxDBClient`
+        mode: `ClientMode`
+            Client mode, async or sync
```

Comment:
Personally, I think that rather than having the mode here be a `ClientMode` enum, which a user has to discover/import and then set, I'd have it be a kwarg (either `async=True` or `synchronous=False`), because not only is that easier for users, but it also means that the new sync mode is optional and non-default, and we can then stay API-compatible with the old code, making the changes backward compatible.

Curious what @rra and @jonathansick think there.

Reply:
Yes, sounds much better for me also.

Reply:
Sorry, I was thinking about this comment, for users should be the same using the ClientMode Enum, because it is not included in the interface is just part of the implementation, in fact all could be done private to prevent any use of it, and also EfdClient is still the async client and have the same arguments as before. Also mode is used in the client by the static base class in the __init_subclass__.

Reply:
Ah right, fair enough. Yeah, it doesn't really matter then, so whatever you prefer. I personally don't like more classes/enum etc when bools are just as good, and there's not really any potential for another mode in the future (you're either async or not), so I can't see how a bool would ever be insufficient, but keeping it as-is is fine too.